### PR TITLE
perf(share): don't broadcast datahash of empty EDS

### LINKED ISSUE #1775: bug(share/p2p/shrexsub | das): don't broadcast empty blocks' data hashes
When dasing over empty blocks, the `BroadcastFn` is still called, broadcasting empty block hashes over shrexsub

### PR DESCRIPTION
**Pull request by Wondertan**
Closes #1775

### CODE REVIEW

**Review by vgonkivs — APPROVED** [review]

File: share/empty.go [hunk 1]
```diff
@@ -13,6 +14,29 @@ import (
 	"github.com/celestiaorg/rsmt2d"
 )
 
+var (
+	emptyRoot atomic.Pointer[Root]
+	emptyEDS  atomic.Pointer[rsmt2d.ExtendedDataSquare]
```
Comment (walldiss) [inline share/empty.go:19]: Why do you need atomic if you only write during init?

File: share/p2p/shrexsub/pubsub.go [hunk 2]
```diff
@@ -100,5 +101,10 @@ func (s *PubSub) Subscribe() (*Subscription, error) {
 
 // Broadcast sends the EDS notification (DataHash) to every connected peer.
 func (s *PubSub) Broadcast(ctx context.Context, data share.DataHash) error {
+	if bytes.Equal(share.EmptyRoot().Hash(), data) {
+		// no need to broadcast datahash of an empty block
+		return nil
+	}
+
```
[the remark this answers is not in the document]
Comment (Wondertan) [reply share/p2p/shrexsub/pubsub.go:108]: Good idea

File: share/empty.go (hunk 1 above)
Comment (Wondertan) [reply share/empty.go:19]: True, originally, I did that lazily with atomict, but now it makes sense to avoid atomic at all

**Review by walldiss — DISMISSED** [review]

**Review by renaynay — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

File: share/empty.go [hunk 3]
```diff
@@ -14,26 +14,27 @@ import (
 )
 
 var (
-	emptyRoot *Root
-	emptyEDS  *rsmt2d.ExtendedDataSquare
+	// EmptyRoot returns Root of an empty EDS.
+	EmptyRoot *Root
+	// EmptyEDS returns the EDS of the empty block data square.
+	EmptyEDS *rsmt2d.ExtendedDataSquare
```
Comment (walldiss) [inline share/empty.go:20]: Previous version with funcs as getters didn't allow to modify values. Now it is possible to write to EmptyRoot, EmptyEDS vars

**Comment by Wondertan** [comment]
Ok, so I simplified some things and fixed potential panic

File: share/empty.go (hunk 3 above)
Comment (Wondertan) [reply share/empty.go:20]: Well, yeah

Comment (Wondertan) [reply share/empty.go]: Better keep it encapsulated

**Review by walldiss — DISMISSED** [review]

**Review by renaynay — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Pull request merged**